.circleci/config.yml

```diff
@@ -46,7 +46,12 @@ jobs:
           key: ccache-debug-linux-x86_64-v3-{{ .Branch }}-{{ epoch }}
           paths:
             - ~/.cache/ccache
-      - run: scripts/clang-tidy-only-diff.sh 4
+      # clang-tidy is quite slow and might take much longer than default 10m limit
+      - run:
+          name: Run clang-tidy checks
+          no_output_timeout: 30m
+          # Limit concurrency to 6 to avoid OOMKills on CircleCI runners.
+          command: scripts/clang-tidy-only-diff.sh 6
```

Comment:
Concurrency increase from 4 to 6 speeds up this step from 30m to 20m on such rare occasions with such huge PRs.